src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -404,49 +405,65 @@ object PhaseBlock extends LazyLogging {
       progress.record(ctx.getContig, ctx.getStart)
     }
 
+    val blockSet = new util.TreeSet[PhaseBlock](new Comparator[PhaseBlock] {
+      def compare(a: PhaseBlock, b: PhaseBlock): Int = {
+        if (a.start < b.start) -1
+        else if (a.start == b.start) b.length - a.length
+        else 1
+      }
+    })
+
+    {
+      import scala.collection.JavaConversions.asJavaCollection
+      blockSet.addAll(phaseBlocks.values)
+    }
+
     // Make sure we do not have any overlapping phase blocks
     // - if block #1 is enclosed in block #2, keep only block #2
     // - otherwise, if block #1 and #2 overlap, truncate the smaller one
     val blocks = ListBuffer[PhaseBlock]()
     logger.info(s"Found ${phaseBlocks.size} phase block")
-    phaseBlocks.values.toList.sortBy { b => (b.start, -b.length) } match {
-      case Nil                    => Unit
-      case block :: Nil           => blocks.append(block)
-      case block :: currentBlocks =>
-        var left = block
-        currentBlocks.foreach { right =>
-          require(left.getStart <= right.getStart)
-          if (left.overlaps(right)) {
-            if (!modifyBlocks) {
-              throw new IllegalStateException(s"Block $left overlaps $right")
-            }
-            if (left.encloses(right)) { // keep the left block only
-              logger.info(s"Removing $right enclosed in $left")
-              // do nothing
-            }
-            else if (right.encloses(left)) { // keep the right block
-              logger.info(s"Removing $left enclosed in $right")
-              left = right
-            }
-            else if (left.length < right.length) { // truncate left
-              logger.info(s"Truncating $left which overlaps $right");
-              blocks.append(left.copy(end=right.getStart-1))
-              left = right
-            }
-            else { // truncate right
-              logger.info(s"Truncating $right which overlaps $left");
-              blocks.append(left)
-              left = right.copy(start=left.getEnd+1)
-            }
-          }
-          else {
-            blocks.append(left)
-            left = right
-          }
+
+    var left = blockSet.pollFirst()
+    while (!blockSet.isEmpty) {
+      val right = blockSet.pollFirst()
+
+      if (right.getStart < left.getStart) {
```

Comment:
Checks like this can be written as just:
`require(left.getStart <= right.getStart, s"left: $left right: $right")`